code/API_definitions/Application_Endpoint_Discovery_API.yaml

```diff
@@ -0,0 +1,596 @@
+      operationId: getAppEndpoints
+      parameters:
+        - name: filter
+          in: query
+          required: true
+          description: |
+            Filter the Edge Cloud Zones according to the parameter value.
+            For this API the only supported value is `closest`.
+          schema:
+            type: string
+            enum:
+              - closest
+        - name: IPv4-Address
+          in: header
+          required: false
+          description: The public IPv4 address allocated to the device.
+          example: "84.125.93.10"
+          schema:
+            $ref: "#/components/schemas/SingleIpv4Addr"
+        - name: IPv6-Address
+          in: header
+          required: false
+          description: The public IPv6 address allocated to the device.
+          example: "2001:db8:85a3:8d3:1319:8a2e:370:7348"
+          schema:
+            $ref: "#/components/schemas/DeviceIpv6Address"
+        - name: Network-Access-Identifier
+          in: header
+          description: |
+            3GPP network access identifier for the subscription
+            being used by the device.
+          example: "4d596ac1-7822-4927-a3c5-d72e1f922c94@domain.com"
+          required: false
+          schema:
+            $ref: "#/components/schemas/NetworkAccessIdentifier"
+        - name: Phone-Number
+          in: header
+          example: "+441234567890"
+          required: true
+          description: |
+            MSISDN in E.164 format (starting with country code) of
+            the mobile subscription being used by the device. Optionally
+            prefixed with '+'.
+          schema:
+            $ref: "#/components/schemas/PhoneNumber"
+        - name: AppId
+          in: header
+          required: true
+          description: |
+            A globally unique identifier associated with the application.
+            OP generates this identifier when the application is submitted over
+            NBI.
+          schema:
+            $ref: "#/components/schemas/AppId"
+        - name: X-Correlator
+          in: header
+          required: false
+          description: |
+            When the API Consumer includes the "X-Correlator" header in the
+            request, the API provider must include it in the response with
+            the same value that was used in the request. Otherwise, it is
+            optional to include the "X-Correlator" header in the response with
+             any valid value. Recommendation is to use UUID for values.
+          schema:
+            type: string
+            format: uuid
+      responses:
+        "200":
+          description: |
+            Successful response, returning the closest App Instance Endpoint
+            Zone to the user device identified in the request header.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/AccessEndpoint"
```

Comment:
As the path itself says /apps-endpoint or if changed to /app-endpoints then shouldnt the AccessEndpoint be also made plural to indicate there could be more than one items depending upon the application design? We should keep options open for application with more than one endpoints while it allows to serve apps with single endpoint too.

Reply:
Agree